sycl/source/device.cpp

```diff
@@ -41,10 +41,20 @@ vector_class<device> device::get_devices(info::device_type deviceType) {
   if (detail::match_types(deviceType, forced_type)) {
     detail::force_type(deviceType, forced_type);
     for (const auto &plt : platform::get_platforms()) {
-      vector_class<device> found_devices(plt.get_devices(deviceType));
-      if (!found_devices.empty())
-        devices.insert(devices.end(), found_devices.begin(),
-                       found_devices.end());
+      // Host device must always be available.
```

Comment:
We can left code like it was before if platform_impl::get_devices() is changed to return host 
device for host platform. To do that you can replace
sycl/source/detail/platform_impl.cpp (lines 196-204)

if (is_host()) 
   return Res.resize(1); // default device constructor creates host device

Reply:
I didn't do that b/c get_devices(something_not_host) is supposed to return nothing in that case.